src/lwc/geBatchWizard/geBatchWizard.js

```diff
@@ -248,11 +254,43 @@ export default class geBatchWizard extends NavigationMixin(LightningElement) {
 
     handleTemplateChange(event) {
         this.selectedTemplateId = event.detail.value;
+        this.selectedBatchHeaderFields = [];
+
+        this.selectedBatchHeaderFields = addKeyToCollectionItems(this.selectedTemplate.batchHeaderFields);
         this.formSections = this.selectedTemplate.layout.sections;
 
         if (this.recordId && this.dataImportBatchRecord && this.dataImportBatchRecord.fields) {
             this.setValuesForSelectedBatchHeaderFields(this.dataImportBatchRecord.fields);
         }
+
+        this.resetValidations();
+    }
+
+    /*******************************************************************************
+    * @description Method collects all batch header fields and checks for their
+    * validity.
+    */
+    validateBatchHeaderFields() {
+        this.resetValidations();
+        this.hasInvalidBatchFields = false;
```

Comment:
I think this line is an unneeded duplicate if you are calling resetValidations() above!